contracts/modules/LoanClosingsBase.sol

```diff
@@ -356,32 +356,55 @@ contract LoanClosingsBase is LoanClosingsShared, LiquidationHelper {
 		uint256 rolloverReward = _getRolloverReward(loanParamsLocal.collateralToken, loanParamsLocal.loanToken, loanLocal.principal);
 
 		if (rolloverReward != 0) {
-			// pay out reward to caller
-			loanLocal.collateral = loanLocal.collateral.sub(rolloverReward);
+			// if the reward > collateral:
+			if (rolloverReward > loanLocal.collateral) {
+				// 1. pay back the excess reward to the lender
+				// 2. pay the complete collateral to msg.sender
+				// 3. close the position & emit close event
+				uint256 excessRewards = rolloverReward.sub(loanLocal.collateral);
+				_closeWithSwap(
+					loanLocal.id,
+					msg.sender,
+					excessRewards,
+					true, /// TODO::should we return it as the collateral or loan token?
+					"" // loanDataBytes
+				);
+			} else {
+				// pay out reward to caller
+				loanLocal.collateral = loanLocal.collateral.sub(rolloverReward);
 
-			_withdrawAsset(loanParamsLocal.collateralToken, msg.sender, rolloverReward);
+				_withdrawAsset(loanParamsLocal.collateralToken, msg.sender, rolloverReward);
+			}
 		}
 
-		//close whole loan if tiny position will remain
-		if (_getAmountInRbtc(loanParamsLocal.loanToken, loanLocal.principal) <= TINY_AMOUNT) {
-			_closeWithDeposit(
-				loanLocal.id,
-				loanLocal.borrower, //TODO loanLocal.borrower ?
-				loanLocal.principal
-			);
-		} else {
-			(uint256 currentMargin, ) =
-				IPriceFeeds(priceFeeds).getCurrentMargin(
-					loanParamsLocal.loanToken,
-					loanParamsLocal.collateralToken,
-					loanLocal.principal,
-					loanLocal.collateral
+		if (loanLocal.collateral > 0) {
```

Comment:
instead of checking the collateral here, you could just place the code below into the else block above - a sit is only to be executed in that context.

Reply:
If we move to the else block above,
If let say the rolloverReward greater than 0, but smaller the collateral, the tiny position check won't be executed then?